tests/suites/test_suite_pkcs7_write.function

```diff
@@ -0,0 +1,154 @@
+{
+    unsigned char *pkcs7_buf = NULL, *new_data = NULL, *key_pem = NULL, *key = NULL, *crt_pem = NULL, *crt = NULL, **crts = NULL, **keys = NULL;
+    size_t pkcs7_len, new_data_len, key_pem_len, key_len, crt_pem_len, crt_len, *crt_sizes = NULL, *key_sizes = NULL;
+    int res, key_pairs;
+    mbedtls_x509_crt *x509 = NULL;
+    mbedtls_pkcs7 pkcs7;
+
+    mbedtls_pkcs7_init( &pkcs7 );
+    key_pairs = 1;
+    res = mbedtls_pkcs7_load_file( data_file, &new_data, &new_data_len );
+    TEST_ASSERT( res == 0 );
+    if( keys_are_signatures )
+    {
+        res = mbedtls_pkcs7_load_file( key_file, &key, &key_len);
+        TEST_ASSERT( res == 0 );
+    }
+    else
+    {
+        res = mbedtls_pkcs7_load_file( key_file, &key_pem, &key_pem_len );
+        TEST_ASSERT( res == 0 );
+        res = mbedtls_convert_pem_to_der( key_pem, key_pem_len, &key, &key_len );
+        TEST_ASSERT( res == 0 );
+    }
+    res = mbedtls_pkcs7_load_file( cert_file, &crt_pem, &crt_pem_len );
+    TEST_ASSERT( res == 0 );
+    res = mbedtls_convert_pem_to_der( crt_pem, crt_pem_len, &crt, &crt_len);
+    TEST_ASSERT( res == 0 );
+
+    ASSERT_ALLOC( keys, key_pairs * sizeof( &key ) );
+    keys[0] = key;
+    ASSERT_ALLOC( crts, key_pairs * sizeof( &crt ) );
+    crts[0] = crt;
+    ASSERT_ALLOC( crt_sizes, key_pairs * sizeof (crt_len) );
+    crt_sizes[0] = crt_len;
+    ASSERT_ALLOC( key_sizes , key_pairs * sizeof (key_len) );
+    key_sizes[0] = key_len;
+
+    /* test that we can correctly create a pkcs7 */
+    res = mbedtls_pkcs7_create(&pkcs7_buf, &pkcs7_len, new_data, new_data_len, (const unsigned char **) crts,(const unsigned char **) keys, crt_sizes, key_sizes, key_pairs, md, mbedtls_test_rnd_pseudo_rand, NULL, keys_are_signatures);
+    TEST_ASSERT( res == 0 );
+
+    /* test that it can be parsed */
+    res = mbedtls_pkcs7_parse_der( pkcs7_buf, pkcs7_len, &pkcs7 );
+    TEST_ASSERT( res == MBEDTLS_PKCS7_SIGNED_DATA );
+
+    /* test that the original data was signed correctly */
+    ASSERT_ALLOC( x509, sizeof( *x509 ) );
+    mbedtls_x509_crt_init ( x509 );
+    res = mbedtls_x509_crt_parse( x509, crts[0], crt_sizes[0] );
+    TEST_ASSERT( res == 0 );
+    res = mbedtls_pkcs7_signed_data_verify( &pkcs7, x509, new_data, new_data_len );
+    TEST_ASSERT( res == 0 );
+
+exit:
+    mbedtls_free( pkcs7_buf );
+    mbedtls_pkcs7_free( &pkcs7 );
+    for( int i = 0; i < key_pairs; i ++ ) {
+        mbedtls_free( crts[i] );
+        mbedtls_free( keys[i] );
+    }
+    mbedtls_free( crts );
+    mbedtls_free( keys );
+    mbedtls_free( new_data );
+    if( !keys_are_signatures )
+    mbedtls_free( key_pem );
+    mbedtls_free( crt_pem );
+    mbedtls_free( key_sizes );
+    mbedtls_free( crt_sizes );
+    mbedtls_x509_crt_free( x509 );
+    mbedtls_free( x509 );
+
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void pkcs7_write_fail( char *data_file, char *cert_file, char *key_file, int md, int keys_are_signatures)
```

Comment:
It looks like this can be combined with the above test by just adding an `expected` argument to give the expected result returned by `mbedtls_pkcs7_create`